src/components/pages/userBooklistsPage/UserBooklistsPage.tsx

```diff
@@ -0,0 +1,224 @@
+
+    const context = useContext(TheContext);
+    const navigate = useNavigate();
+
+    const fetchBooklists = async () => {
+        setBooklists(await fetchUserBooklists());
+    };
+
+    useEffect(() => {
+        fetchBooklists();
+    }, []);
+
+    const handleUserBooksButton = (booklist: Book_list) => {
+        setSelectedBooklist(booklist);
+    }
+
+    useEffect(() => {
+        if (selectedBooklist.id !== -1){
+            setBookListSelected(true);
+        }
+    }, [selectedBooklist]);
+
+    const handleCloseList = () => {
+        setBookListSelected(false);
+        setSelectedBooklist({id:-1, name:''});
+    }
+
+    const renderBookLists = () => {
+        let renderedBooklists = [];
+        for (const booklist of booklists) {
+            renderedBooklists.push(
+                <Paper
+                    elevation={10}
+                    sx={{
+                        padding: "1rem",
+                        width: "60%",
+                        margin: "auto",
+                        marginBottom: 1
+                    }}
+                >
+                    <Stack direction="row" justifyContent="space-between">
+                        <Stack>
+                            <Typography
+                                sx={{
+                                    fontFamily: "Montserrat",
+                                    fontWeight: "bold",
+                                    margin: "auto"
+                                }}
+                            >
+                                {booklist.name}
+                            </Typography>
+                        </Stack>
+                        <Stack marginY={1}>
+                            <Button
+                                sx={listBooksDeleteButton}
+                                variant="contained"
+                                /*
+                                onClick={() => {
+                                    navigate("/userlist");
+                                }}*/
+                                onClick={() => {handleUserBooksButton(booklist)}}
+                            >
+                                View
+                            </Button>
+                            <Button
+                                sx={listBooksLoanButton}
+                                variant="contained"
+                                onClick={async () => {
+                                    await fetchDeleteBooklist(booklist.id);
+                                    await fetchBooklists();
+                                }}
+                            >
+                                Delete
+                            </Button>
+                        </Stack>
+                    </Stack>
+                </Paper>
+            );
+        }
+        return renderedBooklists;
+    };
+
+    
+        /*  using Card component and CardActionArea to click on whole card instead of button component
```

Comment:
Remove this comment or is it going to be used later?

Reply:
Jum suggested that using a card would allow the entire thing to be clicked to open the list. We should ask PO if how they would want it

Reply:
Alright, then its fine :)